### PR TITLE
Booking wrapper

### PR DESCRIPTION
**Pull request by paulhowes-HMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4776

### Change description ###
Create booking wrapper and component placeholders for booking process

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/booking/components/booking-navigation/booking-navigation.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(BookingNavigationComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
Comment (adnanakgun) [inline src/booking/components/booking-navigation/booking-navigation.component.spec.ts:23]: Any chance of testing anything else on this component?

File: src/booking/containers/booking-check/booking-check.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(BookingCheckComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
```
Comment (adnanakgun) [inline src/booking/containers/booking-check/booking-check.component.spec.ts:23]: Any chance of testing anything else on this component?

**Review by tarikdogan-hmcts — DISMISSED** [review]

**Review by adnanakgun — APPROVED** [review]

File: src/booking/components/booking-navigation/booking-navigation.component.spec.ts (hunk 1 above)
Comment (paulhowes-HMCTS) [reply src/booking/components/booking-navigation/booking-navigation.component.spec.ts:23]: I've had a look at the code and other examples where similar navigation is being used. The component currently has 95.65% coverage and I'd only be adding superfluous tests to see if a state is in a list of states or if a trigger is emitted when I call the trigger method.

File: src/booking/components/booking-navigation/booking-navigation.component.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  public bookingNavigationEvent = BookingNavigationEvent;
+
+  public homeVisibilityStates = bookingHomeVisibilityStates;
+  public locationVisibilityStates = bookingLocationVisibilityStates;
+  public dateVisibilityStates = bookingDateVisibilityStates;
+  public checkBookingVisibilityStates = bookingCheckVisibilityStates;
+
+  public backVisibilityStates = bookingBackButtonVisibilityStates;
+
+  public ngOnInit() {
```
[the remark this answers is not in the document]
Comment (paulhowes-HMCTS) [reply src/booking/components/booking-navigation/booking-navigation.component.ts]: Removed as not required.

**Review by tarikdogan-hmcts — APPROVED** [review]

**Review by adnanakgun — APPROVED** [review]

File: src/booking/containers/booking-wrapper/booking-wrapper.component.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  public onNavEvent(event: BookingNavigationEvent) {
+    this.navEvent = {
+      event,
+      timestamp: Date.now()
+    };
+    this.navigationHandler(event);
+  }
+
+  public navigationHandler(navEvent: BookingNavigationEvent) {
+    switch (navEvent) {
```
Comment (tarikdogan-hmcts) [inline src/booking/containers/booking-wrapper/booking-wrapper.component.ts:38]: for now it s ok but we break Open closed  and Seperation of Concerns , we'll finetune later by applying strategy pattern .

**Review by tarikdogan-hmcts — APPROVED** [review]

**Review by adnanakgun — APPROVED** [review]

**Pull request merged**